src/main/java/org/toradocu/translator/CommentTranslator.java

```diff
@@ -3,22 +3,17 @@
 import org.toradocu.extractor.BlockTag;
 import org.toradocu.extractor.DocumentedExecutable;
 import org.toradocu.translator.preprocess.PreprocessorFactory;
-import org.toradocu.translator.spec.Specification;
 
 public class CommentTranslator {
 
-  public static <T extends BlockTag<Specification>> void translate(
-      T tag, DocumentedExecutable excMember) {
+  public static <T extends BlockTag> void translate(T tag, DocumentedExecutable excMember) {
```

Comment:
My suggestion would be to get rid of the `Translate` abstraction, and build translation methods that return the correct type of specification for the structure being translated.

You are close to this already, but are unnecessarily constrained by the `Translate` interface.